### PR TITLE
Adding marks to sliders when the height allows it

### LINKED ISSUE #339: Add ticks to sliders
https://github.com/silx-kit/h5web/pull/335#pullrequestreview-538350630

Or marks as react-slider calls them: https://github.com/zillow/react-slider/blob/master/CHANGELOG.md#110-2020-11-03.

Available since react-slider 1.1.0.

### PR DESCRIPTION
**Pull request by loichuder**
Fix #339

Adds marks only when the slider values are separated by 25px (thumb size). This allows to display marks where the thumb will "stick" (few possible values) and to not display them when moving the thumb feel continuous (many possible values). 

For example, see below: the left slider has 9 possible values and displays marks while the right slider has 41 possible values and therefore do not display them. **Two aspects possible to choose from:**
image
vs.
image

I may have overthought this but I felt the marks should be informative in some way.

### CODE REVIEW

**Comment by loichuder** [comment]
Waiting for https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49920 to update `@types/react-slider` to 1.1

File: src/h5web/dimension-mapper/SlicingSlider.module.css [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+.thumb:active {
+  cursor: none;
+}
+
+.mark {
+  position: absolute;
+  background-color: var(--primary-bg);
+  border: 2px solid var(--secondary-dark);
+  border-radius: 50%;
+  margin: calc(var(--thumb-height) / 2 - var(--mark-height) / 2) 0;
```
Comment (loichuder) [inline src/h5web/dimension-mapper/SlicingSlider.module.css:59]: This is to center the first/last mark with respect to the thumb.

File: src/h5web/dimension-mapper/SlicingSlider.module.css [hunk 2]
```diff
@@ -0,0 +1,62 @@
+:root {
+  --thumb-height: 1.5rem;
+  --mark-height: 0.4rem;
```
[the remark this answers is not in the document]
Comment (axelboc) [reply src/h5web/dimension-mapper/SlicingSlider.module.css:3]: For `--mark-height`, I would use exactly `0.4375rem` (7px) -- i.e. an odd number, for better alignment. (`0.4rem` rounds down to 6px.)

**Review by axelboc — APPROVED** [review]

**Comment by loichuder** [comment]
The committee opted for the second aspect.

**Pull request merged**